### DISCUSSION TITLE
What is the best way to design top form step progress bar with react-hooks-form

### DISCUSSION THREAD

**Discussion by ajaybeniwal** [react-hook-form/react-hook-form#1976] [Q&A] [answered]
We are designing a complex form which has a stepper progress bar associated with various sections it should be ticked when all mandatory fields are selected and should turn into red if there is any ui validation error in that section m so what could be most performant and best way to design such solution 
<img width="1425" alt="Screenshot 2020-06-27 at 12 12 01 PM" src="https://user-images.githubusercontent.com/658004/85914487-cf2e7e00-b870-11ea-838d-32cde0d2808b.png">

**Comment by keiya01** [comment, marked as answer]
I think you can use `useFormContext`.
https://react-hook-form.com/api#useFormContext

**Reply by ajaybeniwal** [reply]
Question is not about how to access the form but how to build in way that rerenders are less and the form is more performant

**Reply by keiya01** [reply]
What do you want to do? I want to know what you are hoping for explicitly.
Do you want to manage validation timing ?

**Reply by keiya01** [reply]
If you want to do only validation in each section, you can use `triggerValidation`.
https://react-hook-form.com/api#triggerValidation

**Reply by ajaybeniwal** [reply]
Suppose there is one section which has 3 fields are 2 fields are mandatory so when both these 2 fields are filled the progress bar should turn to ticked and if any of these 3 fields have any error the progress status should turn to red

**Reply by keiya01** [reply]
If you have no idea about resolving this issue, I can not suggest explicit resolution. 
I can suggest react-hook-form's API for your solution.

**Reply by bluebill1049** [reply]
There are multiple ways to do multiple steps funnel: here is a video tutorial which you may find useful: https://www.youtube.com/watch?v=CeAkxVwsyMU

**Reply by bluebill1049** [reply]
here is another example which was done by our user: https://github.com/react-hook-form/react-hook-form/discussions/1979
